commit af4a740c24a611fe3f18df4bd5fa85356b04cdba (refs/changes/16/1716/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-03-27T18:40:40+00:00 (2 years, 6 months ago)
    
    MANTA-3203 binder serving NOTIMP responses on empty services
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Robert Mustacchi <rm@joyent.com>

diff --git a/lib/server.js b/lib/server.js
index 2d214e8..249796a 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -261,7 +261,12 @@ function resolve(options, query, cb) {
                                         query.setError('nxdomain');
                                         break;
                                 }
-                                // Inefficient, but easy to reason about.
+                                /*
+                                 * Make sure we set noerror here, otherwise we
+                                 * would respond with NOTIMP to a query about
+                                 * a service with no children.
+                                 */
+                                query.setError('noerror');
                                 var recs = record.children.filter(
                                     function (sub) {
                                         return (sub.type === 'load_balancer' ||
